s4e-backend/src/main/java/pl/cyfronet/s4e/service/SearchService.java

```diff
@@ -43,6 +58,32 @@ public Long getCountBy(Map<String, Object> params) throws SQLException, QueryExc
         return repository.countAllByParamsMap(params);
     }
 
+    public void checkProductTypePresent(Map<String, Object> params) throws BadRequestException {
+        if (params.get(PRODUCT_TYPE) == null) {
+            throw new BadRequestException("No product type.");
+        }
+    }
+
+    public void applyLicenseByProductType(Map<String, Object> params) throws NotFoundException {
+        val product =
+                productRepository.findByName(
+                        String.valueOf(params.get(PRODUCT_TYPE)), Product.class)
+                        .orElseThrow(() -> new NotFoundException("Product not found"));
+        val accessType = product.getAccessType();
+        AppUserDetails userDetails = AppUserDetailsSupplier.get();
+        if (Product.AccessType.EUMETSAT.equals(accessType)) {
+            // if user doesnt have proper license add parameter to get older scenes
+            if (!userDetails.getAuthorities().contains(new SimpleGrantedAuthority("LICENSE_EUMETSAT"))) {
+                params.put(SearchQueryParams.ACCESS_TYPE, Product.AccessType.EUMETSAT);
+            }
+        } else if (Product.AccessType.PRIVATE.equals(accessType)) {
+            // if user doesnt have proper license throw 403
+            if (!licensePermissionEvaluator.allowProductRead(product.getId(), userDetails)) {
+                throw new AccessDeniedException("Access Denied");
+            }
```

Comment:
Do this check regardless of the license type, before even checking for the EUMETSAT type